### DISCUSSION TITLE
Cannot add chezmoi file to chezmoi (/Users/.../.config/chezmoi/chezmoistate.boltdb is protected)

### DISCUSSION THREAD

**Discussion by Demianeen** [twpayne/chezmoi#3645] [Q&A] [answered]
I try to add ~/.config folder to chermoi but get the error above. It seams like chezmoi don't want to add its state.

How do you add ~/.config folder? Or do you add each dotconfig manually?

**Comment by twpayne** [comment, marked as answer]
You have to add each folder in `~/.config` manually.

**Reply by Lockszmith-GH** [reply]
if you want to do it in bulk, you can use `xargs`:

```bash
echo ~/.config/* | xargs -rn1 chezmoi add 
```

But keep in mind, you DON'T want to add chezmoi's config dir

So the following might be safer:

```bash
echo ~/.config/* | xargs -rn1 | grep -v chezmoi | xargs -rn1 chezmoi add 
```

**Reply by Demianeen** [reply]
Thanks, for the answer @twpayne and command @Lockszmith-GH! 

@twpayne, can you explain why does `.chezmoiignore` not work with chezmoi folder? Is this limitation intended? Or is it related to how the chezmoi ignore works?

**Reply by twpayne** [reply]
`chezmoi add` currently checks `.chezmoiignore` after checking for protected paths like `~/.config/chezmoi`. This will be fixed when #3630 is fixed.
